### PR TITLE
Jaeger spans for availability distribution

### LINKED ISSUE #2551: Add back Jaeger spans to availability distribution
Follow up for #2423 .

### PR DESCRIPTION
**Pull request by eskimor**
Closes #2551

- Introduces logging wrapper for mick_jaeger logging facility.
- Also fixes instrumentation macros to use log target properly.

### CODE REVIEW

**Comment by ordian** [comment]
cc @tomaka for jaeger things

File: node/network/availability-distribution/src/requester/fetch_task/mod.rs [hunk 1]
```diff
@@ -254,8 +261,14 @@ impl RunningTask {
 		let mut bad_validators = Vec::new();
 		let mut label = FAILED;
 		let mut count: u32 = 0;
+		let mut _span = self.span.child_builder("fetch-task")
+			.with_chunk_index(self.request.index.0)
+			.with_relay_parent(&self.relay_parent)
+			.build();
+		let mut span_log = _span.log();
```
Comment (drahnr) [inline node/network/availability-distribution/src/requester/fetch_task/mod.rs:270]: As discussed, we want to stick to tags and the builder API for now.

File: node/jaeger/src/lib.rs [hunk 2]
```diff
@@ -331,6 +339,33 @@ impl Span {
...
+	/// Just a wrapper around `mick_jaeger::Log::with_string`.
+	pub fn with_string(self, key: &str, value: &str) -> Self {
+		if let Some(log) = self.inner {
+			Self{inner: Some(log.with_string(key, value))}
+		} else {
+			self
+		}
+	}
+}
+
```
Comment (drahnr) [inline node/jaeger/src/lib.rs:368]:
```suggestion
```

File: node/jaeger/src/lib.rs [hunk 3]
```diff
@@ -321,6 +321,14 @@ impl Span {
...
 	}
 
+	/// Add logs to this span.
+	pub fn log(&mut self) -> Log<'_> {
+		match self {
+			Self::Enabled(inner) => Log { inner: Some(inner.log()) },
+			Self::Disabled => Log { inner: None },
+		}
+	}
+
```
Comment (drahnr) [inline node/jaeger/src/lib.rs:331]:
```suggestion

```

**Review by drahnr — COMMENTED** [review]
A few nits about using int tags rather than logs, other than that looks good, we might want to have a few more child span to get a bit more detail for `dot-jaeger`.

Does `fields(target = ` have side effects for the consumer of the `tracing::Span`s?

File: node/network/availability-distribution/src/responder.rs [hunk 4]
```diff
@@ -64,6 +64,12 @@ pub async fn answer_request<Context>(
 where
 	Context: SubsystemContext,
 {
+	let mut span = jaeger::candidate_hash_span(&req.payload.candidate_hash, "answer_request");
+	span.add_stage(jaeger::Stage::AvailabilityDistribution);
+	let _child_span = span.child_builder("answer_chunk_request")
```
Comment (ordian) [inline node/network/availability-distribution/src/responder.rs:69]: we're using kebab-case elsewhere, right?
```suggestion
	let mut span = jaeger::candidate_hash_span(&req.payload.candidate_hash, "answer-request");
	span.add_stage(jaeger::Stage::AvailabilityDistribution);
	let _child_span = span.child_builder("answer-chunk-request")
```

**Review by ordian — APPROVED** [review]

File: node/jaeger/src/lib.rs [hunk 5]
```diff
@@ -313,6 +313,14 @@ impl Span {
 		}
 	}
 
+	/// Add an additional int tag to the span.
+	pub fn add_int_tag(&mut self, tag: &str, value: i64) {
```
Comment (drahnr) [inline node/jaeger/src/lib.rs:317]: :+1:

**Review by drahnr — APPROVED** [review]

**Comment by eskimor** [comment]
bot merge

**Pull request merged**